docs/gc.md

```diff
@@ -31,35 +31,57 @@ The Allocator assigns areas of the Java heap for Java objects. See [Memory alloc
+OpenJ9 provides several GC policies that are designed around different application workloads. Each GC policy consists of a set of characteristics and features that aim to optimize one or more performance aspects of a running application. These performance aspects include application throughput, memory footprint, average pause times, worst-case pause times, and startup time.
 
-| GC Policy                                        | Heap Configuration              | Collectors                         |
-|--------------------------------------------------|---------------------------------|------------------------------------|
-| [`gencon     `](#gencon-policy-default) (default)| Generational (nursery/tenure)   | Scavenger, Concurrent Mark, Global |
-| [`balanced   `](#balanced-policy)                | Generational (multiple regions) | Incremental Generational           |
-| [`metronome  `](#metronome-policy)               | Segregated                      | Realtime                           |
-| [`optavgpause`](#optavgpause-policy)             | Flat                            | Concurrent Mark, Global            |
-| [`optthruput `](#optthruput-policy)              | Flat                            | Global                             |
-| [`nogc       `](#nogc-policy)                    | --                              | --                                 |
+Different policies require a Java heap that is configured in different ways in order to achieve different goals. The simplest configuration consists of a single area of memory, often referred to as a *flat* heap. Other configurations divide the heap into different areas or regions, which might contain objects of different ages (*generations*) or sizes.
 
-All OpenJ9 GC policies support compressed references on 64-bit platforms, which compresses heap pointers to 32 bits if the total heap size does not exceed 63 GB. Applications that require more heap space can forgo compressed pointers and select any heap size within the bounds imposed by the operating system and available system RAM.
+A GC cycle is a repeatable process that involves a set of GC operations. These operations process all or parts of the Java heap to complete a discrete function. For example, a *mark* operation traces all objects in the heap to determine which ones are reachable. A *sweep* operation runs to clear away unreachable objects. Together, a *mark* and *sweep*  operation are capable of reclaiming used memory as part of a GC cycle.
+
+A GC operation might complete in one step, or it might involve phases. For example, a *mark* operation consists of the following 3 phases:
+
+- *Initial phase:* Identify all the root objects by running a root scan.
+- *Main phase:* Trace references from each root object to other objects in the heap. Reachable objects are marked.
```

Comment:
Consider this (recursively is somewhat nice to use here, since most objects are only indirectly, through several hops of references, reachable from roots):
From identified initial set of reachable objects, continue recursively tracing heap object graph (or just: tracing references). Reachable objects are marked.

Reply:
Updated